packages/app-mobile/components/ScreenHeader/index.tsx

```diff
@@ -678,8 +690,9 @@ class ScreenHeaderComponent extends PureComponent<ScreenHeaderProps, ScreenHeade
 			);
 
 		return (
-			<View style={this.styles().container}>
-				<View style={{ flexDirection: 'row', alignItems: 'center' }}>
+			<View style={this.styles().outerContainer}>
+				<View style={this.styles().aboveHeader}/>
+				<View style={this.styles().innerContainer}>
```

Comment:
This pull request works around the issue by covering the upper part of the shadow with a `<View/>` (as I believe was the case prior to https://github.com/laurent22/joplin/pull/12785).

There may be [more options for changing the box shadow style](https://reactnative.dev/docs/view-style-props#boxshadow) after Joplin upgrades to the New Architecture.